cmd/api/main.go

```diff
@@ -5,41 +5,35 @@ package main
 
 import (
 	"fmt"
-	"log"
-	"net/http"
-
+	"github.com/kubearchive/kubearchive/cmd/api/auth"
+	"github.com/kubearchive/kubearchive/cmd/api/routers"
 	"github.com/kubearchive/kubearchive/pkg/observability"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"log"
 
 	"github.com/gin-gonic/gin"
 	"go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin"
 )
 
-// FIXME This will be taken from a shared pkg with sink based on the DB schema
-// Just for the first approach
-type resources struct {
-	Kind       string   `json:"kind"`
-	APIVersion string   `json:"apiVersion"`
-	Items      []string `json:"items"`
-}
-
-// getAllResources responds with the list of resources of a specific type across all namespaces
-func getAllResources(c *gin.Context) {
-	group := c.Param("group")
-	version := c.Param("version")
-	pluralResourceType := c.Param("resourceType")
-	response := resources{
-		// FIXME The plural of the resource type can be different of adding an `s` to the kind
-		Kind:       pluralResourceType[:len(pluralResourceType)-1],
-		APIVersion: fmt.Sprintf("%s/%s", group, version),
-		Items:      []string{"resource1", "resource2"},
+var k8sClient = func() *kubernetes.Clientset {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(fmt.Sprintf("Error retrieving in-cluster k8s client config: %s", err.Error()))
 	}
-	c.IndentedJSON(http.StatusOK, response)
-}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(fmt.Sprintf("Error instantiating k8s from host %s: %s", config.Host, err.Error()))
+	}
+	return client
+}()
```

Comment:
Should these parentheses be here? I don't recognize this syntax.

Reply:
This parenthesis are for calling the function when it's declared.
This is called IIF (Immediately Invoked Function), a special usage of function.

You can read more about its usage [here](https://golangbyexample.com/function-golang-complete-guide/#IIF_or_Immediately_Invoked_Function)